From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 20 May 2023 18:52:32 +0100
Subject: [PATCH] Fixup JukeAlert

Use removeIf in GUICommand.

Remove redundant saveDefaultConfig()

diff --git a/build.gradle.kts b/build.gradle.kts
new file mode 100644
index 0000000000000000000000000000000000000000..0f884e8e7546e3c3f4783dac2de5de05411d8f28
--- /dev/null
+++ b/build.gradle.kts
@@ -0,0 +1,13 @@
+group = "civ.genre.jukealert"
+version = "3.0.5"
+description = "JukeAlert"
+
+configure<net.civmc.civgradle.CivGradleExtension> {
+	pluginName = project.property("description") as String
+}
+
+dependencies {
+	compileOnly(project(":CivModCore"))
+	compileOnly(project(":NameLayer"))
+	compileOnly(project(":Citadel"))
+}
diff --git a/src/main/java/com/untamedears/jukealert/JukeAlert.java b/src/main/java/com/untamedears/jukealert/JukeAlert.java
index 1cc38740144009fc2e5138b81867cc9cf873be9a..a2089e472dcc69d91c1385a93e885fa325ac5971 100644
--- a/src/main/java/com/untamedears/jukealert/JukeAlert.java
+++ b/src/main/java/com/untamedears/jukealert/JukeAlert.java
@@ -84,7 +84,6 @@ public class JukeAlert extends ACivMod {
 		snitchConfigManager = new SnitchTypeManager();
 		cullManager = new SnitchCullManager();
 		configManager = new JAConfigManager(this, snitchConfigManager);
-		saveDefaultConfig();
 		dao = new JukeAlertDAO(configManager.getDatabase(getConfig()));
 		if (!dao.updateDatabase()) {
 			getLogger().severe("Errors setting up database, shutting down");
diff --git a/src/main/java/com/untamedears/jukealert/commands/GUICommand.java b/src/main/java/com/untamedears/jukealert/commands/GUICommand.java
index 4dd79e31c64ececfc3c2f435f88c24dad6bf3c87..090cfa53f3ff865c5a1faff4ef1dfc812e760ba0 100644
--- a/src/main/java/com/untamedears/jukealert/commands/GUICommand.java
+++ b/src/main/java/com/untamedears/jukealert/commands/GUICommand.java
@@ -11,7 +11,6 @@ import com.untamedears.jukealert.util.JAUtility;
 import com.untamedears.jukealert.util.JukeAlertPermissionHandler;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Iterator;
 import org.bukkit.ChatColor;
 import org.bukkit.entity.Player;
 
@@ -30,13 +29,7 @@ public class GUICommand extends BaseCommand {
 		Collection<Snitch> snitches = JukeAlert.getInstance().getSnitchManager()
 				.getSnitchesCovering(player.getLocation());
 		// Sort out the ones the player has no perms for
-		Iterator<Snitch> iter = snitches.iterator();
-		while (iter.hasNext()) {
-			Snitch snitch = iter.next();
-			if (!snitch.hasPermission(player, JukeAlertPermissionHandler.getReadLogs())) {
-				iter.remove();
-			}
-		}
+		snitches.removeIf(snitch -> !snitch.hasPermission(player, JukeAlertPermissionHandler.getReadLogs())); // CivPatched
 		if (snitches.isEmpty()) {
 			player.sendMessage(
 					ChatColor.RED + "You do not own any snitches nearby or lack permission to view their logs!");
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
index 0e1a39cc4e0fde0413162812f3bbd55f44973135..df44043e5b048c4dd84fda5c20915ed9cc7258be 100644
--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -4,7 +4,7 @@ version: ${version}
 authors: [mgrandi, Maxopoly]
 depend: [CivModCore, NameLayer, Citadel]
 softdepend: [VanishNoPacket]
-api-version: 1.16
+api-version: 1.18
 description: JukeAlert provides an easy-to-use means by which one can record player activity within an 11 block radius. Anything
    someone does while within eleven blocks of a jukebox you've reinforced will be logged on it, and can be read by looking
    at the jukebox and typing '/jainfo'.
