From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 20 May 2023 18:52:32 +0100
Subject: [PATCH] Remove redundant saveDefaultConfig()


diff --git a/src/main/java/com/untamedears/jukealert/JukeAlert.java b/src/main/java/com/untamedears/jukealert/JukeAlert.java
index 1cc38740144009fc2e5138b81867cc9cf873be9a..a2089e472dcc69d91c1385a93e885fa325ac5971 100644
--- a/src/main/java/com/untamedears/jukealert/JukeAlert.java
+++ b/src/main/java/com/untamedears/jukealert/JukeAlert.java
@@ -84,7 +84,6 @@ public class JukeAlert extends ACivMod {
 		snitchConfigManager = new SnitchTypeManager();
 		cullManager = new SnitchCullManager();
 		configManager = new JAConfigManager(this, snitchConfigManager);
-		saveDefaultConfig();
 		dao = new JukeAlertDAO(configManager.getDatabase(getConfig()));
 		if (!dao.updateDatabase()) {
 			getLogger().severe("Errors setting up database, shutting down");
