From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Fri, 19 May 2023 21:48:51 +0100
Subject: [PATCH] Fixup CivChat2


diff --git a/build.gradle.kts b/build.gradle.kts
index 395cbef720ada467aa901b82ca443181c3effe7c..2483a1156dfea3cd1ba2e7f628cf56683295f4f6 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,25 +1,12 @@
-import net.civmc.civgradle.CivGradleExtension
+group = "civ.genre.civchat2"
+version = "2.0.3"
+description = "CivChat2"
 
-plugins {
-	id("net.civmc.civgradle") version "2.+" apply false
+configure<net.civmc.civgradle.CivGradleExtension> {
+	pluginName = project.property("description") as String
 }
 
-subprojects {
-	apply(plugin = "java-library")
-	apply(plugin = "maven-publish")
-	apply(plugin = "net.civmc.civgradle")
-
-	configure<CivGradleExtension> {
-		pluginName = project.property("pluginName") as String
-	}
-
-	repositories {
-		mavenCentral()
-		maven("https://repo.civmc.net/repository/maven-public")
-		maven("https://repo.aikar.co/content/groups/aikar/")
-		maven("https://libraries.minecraft.net")
-		maven("https://repo.codemc.io/repository/maven-public/")
-
-		maven("https://jitpack.io")
-	}
+dependencies {
+	compileOnly(project(":CivModCore"))
+	compileOnly(project(":NameLayer"))
 }
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
index 3264cd0e2799d08eb52530856f6ba33ae821d5e4..d949b4dd2e8443910ef676144c54519f8437f8fc 100644
--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -1,7 +1,7 @@
 name: ${name}
 main: vg.civcraft.mc.civchat2.CivChat2
 version: ${version}
-api-version: 1.16
+api-version: 1.18
 authors:
 - jjj5311
 - Maxopoly
