From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Fri, 19 May 2023 21:48:51 +0100
Subject: [PATCH] Fixup CivChat2


diff --git a/build.gradle.kts b/build.gradle.kts
new file mode 100644
index 0000000000000000000000000000000000000000..d5071db1a086e307e328ddb6022244f478239e0c
--- /dev/null
+++ b/build.gradle.kts
@@ -0,0 +1,12 @@
+group = "civ.genre.civchat2"
+version = "2.1.0"
+description = "CivChat2"
+
+configure<net.civmc.civgradle.CivGradleExtension> {
+	pluginName = project.property("description") as String
+}
+
+dependencies {
+	compileOnly(project(":CivModCore"))
+	compileOnly(project(":NameLayer"))
+}
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
index 3264cd0e2799d08eb52530856f6ba33ae821d5e4..d949b4dd2e8443910ef676144c54519f8437f8fc 100644
--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -1,7 +1,7 @@
 name: ${name}
 main: vg.civcraft.mc.civchat2.CivChat2
 version: ${version}
-api-version: 1.16
+api-version: 1.18
 authors:
 - jjj5311
 - Maxopoly
