From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Fri, 19 May 2023 21:51:11 +0100
Subject: [PATCH] Yeet sendOwnKills setting

Why this was ever added is beyond comprehension.

diff --git a/src/main/java/vg/civcraft/mc/civchat2/listeners/KillListener.java b/src/main/java/vg/civcraft/mc/civchat2/listeners/KillListener.java
index 908b910751ae96a68951c9e1ee992ca9334451eb..0f42a67ec2fc89dc29293c34722fd58213a489a0 100644
--- a/src/main/java/vg/civcraft/mc/civchat2/listeners/KillListener.java
+++ b/src/main/java/vg/civcraft/mc/civchat2/listeners/KillListener.java
@@ -35,9 +35,6 @@ public class KillListener implements Listener {
 			return;
 		}
 		Player killer = victim.getKiller();
-		if (!settingsMan.getSendOwnKills(killer.getUniqueId())) {
-			return;
-		}
 		String msg;
 		ItemStack item = killer.getInventory().getItemInMainHand();
 		String victimFormattedName = String.format("%s%s", ChatColor.ITALIC, victim.getDisplayName());
diff --git a/src/main/java/vg/civcraft/mc/civchat2/utility/CivChat2SettingsManager.java b/src/main/java/vg/civcraft/mc/civchat2/utility/CivChat2SettingsManager.java
index 9e61152617fccfcd8e509e6ea2679c5d8ef3a17d..65096f554a275d1c489afc37d39833b15ee37523 100644
--- a/src/main/java/vg/civcraft/mc/civchat2/utility/CivChat2SettingsManager.java
+++ b/src/main/java/vg/civcraft/mc/civchat2/utility/CivChat2SettingsManager.java
@@ -15,7 +15,6 @@ public class CivChat2SettingsManager {
 
 	private BooleanSetting showJoins;
 	private BooleanSetting showLeaves;
-	private BooleanSetting sendOwnKills;
 	private BooleanSetting receiveKills;
 	private BooleanSetting receiveKillsFromIgnoredPlayers;
 	private BooleanSetting showChatGroup;
@@ -39,10 +38,6 @@ public class CivChat2SettingsManager {
 				"Should player leave messages be shown?");
 		PlayerSettingAPI.registerSetting(showLeaves, menu);
 
-		sendOwnKills = new BooleanSetting(CivChat2.getInstance(), true, "Broadcast your kills", "civChatBroadcastKills",
-				"Should kills you make be broadcasted to nearby players?");
-		PlayerSettingAPI.registerSetting(sendOwnKills, menu);
-
 		receiveKills = new BooleanSetting(CivChat2.getInstance(), true, "Receive kill broadcasts",
 				"civChatReceiveKills", "Do you want to receive broadcasts for nearby kills");
 		PlayerSettingAPI.registerSetting(receiveKills, menu);
@@ -79,10 +74,6 @@ public class CivChat2SettingsManager {
 		return showLeaves.getValue(uuid);
 	}
 	
-	public boolean getSendOwnKills(UUID uuid) {
-		return sendOwnKills.getValue(uuid);
-	}
-	
 	public boolean getReceiveKills(UUID uuid) {
 		return receiveKills.getValue(uuid);
 	}
