From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 24 May 2023 02:56:25 +0100
Subject: [PATCH] Fix BeeKeeping NMS javadoc reference


diff --git a/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/BeeKeeping.java b/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/BeeKeeping.java
index 34a140bcacdf4f97563d9b496569a60bd6a9cf9c..ac21eedbe579add2d645f1f6bc718c77d389f9db 100644
--- a/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/BeeKeeping.java
+++ b/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/BeeKeeping.java
@@ -45,6 +45,7 @@ import vg.civcraft.mc.civmodcore.chat.ChatUtils;
 import vg.civcraft.mc.civmodcore.inventory.items.ItemUtils;
 
 // CivPatched Start
+import net.minecraft.world.entity.NeutralMob;
 import org.bukkit.ChatColor;
 // CivPatched End
 
@@ -102,7 +103,7 @@ public final class BeeKeeping extends BasicHack {
 		}
 		final Bee bee = (Bee) ((CraftEntity) attacker).getHandle();
 		// Undo bee pacification and allow the bee to sting again
-		/** See the code in {@link IEntityAngerable#pacify()} */
+		/** See the code in {@link NeutralMob#stopBeingAngry()} */
 		final var lastDamageCause = bee.getLastHurtByMob();
 		final var angerTarget = bee.getPersistentAngerTarget();
 		final var goalTarget = bee.getTarget();
