From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 14 Jun 2023 07:56:29 +0100
Subject: [PATCH] Add asAudience to Group


diff --git a/src/main/java/vg/civcraft/mc/namelayer/group/Group.java b/src/main/java/vg/civcraft/mc/namelayer/group/Group.java
index e15133b9393741b1645fd1bd81207ec508fffb49..d6f99d624a284cbd418e22a37c032248629ad7a1 100644
--- a/src/main/java/vg/civcraft/mc/namelayer/group/Group.java
+++ b/src/main/java/vg/civcraft/mc/namelayer/group/Group.java
@@ -17,7 +17,11 @@ import vg.civcraft.mc.namelayer.NameLayerPlugin;
 import vg.civcraft.mc.namelayer.database.GroupManagerDao;
 
 // CivPatched Start
+import java.util.Objects;
+import net.kyori.adventure.audience.Audience;
+import net.kyori.adventure.audience.ForwardingAudience;
 import net.kyori.adventure.text.Component;
+import org.bukkit.Bukkit;
 import org.jetbrains.annotations.NotNull;
 // CivPatched End
 
@@ -96,6 +100,16 @@ public class Group {
 		}
 	}
 	
+	// CivPatched Start
+	public @NotNull ForwardingAudience asAudience() {
+		return () -> this.players.keySet()
+				.stream()
+				.map((playerUUID) -> (Audience) Bukkit.getPlayer(playerUUID))
+				.filter(Objects::nonNull)
+				.toList();
+	}
+	// CivPatched End
+
 	/**
 	 * Returns all the uuids of the members in this group.
 	 * @return Returns all the uuids.
