From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 31 May 2023 01:20:35 +0100
Subject: [PATCH] Fixup CivModCore


diff --git a/build.gradle.kts b/build.gradle.kts
index 59aa4b7b2ea86655902bd5b5e140e77003b8e65a..1f7991da7e8c7fa084a8e10c46d8718c7a018529 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,23 +1,15 @@
-import net.civmc.civgradle.CivGradleExtension
+group = "civ.genre.civmodcore"
+version = "2.3.5"
+description = "CivModCore"
 
-plugins {
-	id("net.civmc.civgradle") version "2.+" apply false
+configure<net.civmc.civgradle.CivGradleExtension> {
+	pluginName = project.property("description") as String
 }
 
-subprojects {
-	apply(plugin = "java-library")
-	apply(plugin = "maven-publish")
-	apply(plugin = "net.civmc.civgradle")
-
-	configure<CivGradleExtension> {
-		pluginName = project.property("pluginName") as String
-	}
-
-	repositories {
-		mavenCentral()
-        maven("https://repo.aikar.co/content/groups/aikar/")
-        maven("https://libraries.minecraft.net")
-
-		maven("https://jitpack.io")
-	}
+dependencies {
+	api("com.zaxxer:HikariCP:5.0.1")
+	api("co.aikar:acf-bukkit:0.5.0-SNAPSHOT")
+	api("co.aikar:taskchain-bukkit:3.7.2")
+	api("com.github.IPVP-MC:canvas:91ec97f076")
+	api("org.apache.commons:commons-collections4:4.4")
 }
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java b/src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java
index b42c2547061518f92bafe8438fab105213510095..146c7b181acdc67f5eb375644db485af384a5e0a 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java
@@ -10,13 +10,11 @@ import vg.civcraft.mc.civmodcore.commands.CommandManager;
 import vg.civcraft.mc.civmodcore.commands.StatCommand;
 import vg.civcraft.mc.civmodcore.dao.DatabaseCredentials;
 import vg.civcraft.mc.civmodcore.dao.ManagedDatasource;
-import vg.civcraft.mc.civmodcore.events.CustomEventMapper;
 import vg.civcraft.mc.civmodcore.inventory.gui.ClickableInventoryListener;
 import vg.civcraft.mc.civmodcore.inventory.items.EnchantUtils;
 import vg.civcraft.mc.civmodcore.inventory.items.MoreTags;
 import vg.civcraft.mc.civmodcore.inventory.items.SpawnEggUtils;
 import vg.civcraft.mc.civmodcore.inventory.items.TreeTypeUtils;
-import vg.civcraft.mc.civmodcore.maps.MapColours;
 import vg.civcraft.mc.civmodcore.players.scoreboard.bottom.BottomLineAPI;
 import vg.civcraft.mc.civmodcore.players.scoreboard.side.ScoreBoardAPI;
 import vg.civcraft.mc.civmodcore.players.scoreboard.side.ScoreBoardListener;
@@ -29,7 +27,6 @@ import vg.civcraft.mc.civmodcore.world.locations.chunkmeta.api.ChunkMetaAPI;
 import vg.civcraft.mc.civmodcore.world.locations.chunkmeta.stat.LoadStatisticManager;
 import vg.civcraft.mc.civmodcore.world.locations.global.CMCWorldDAO;
 import vg.civcraft.mc.civmodcore.world.locations.global.WorldIDManager;
-import vg.civcraft.mc.civmodcore.world.operations.ChunkOperationManager;
 
 public final class CivModCorePlugin extends ACivMod {
 
@@ -77,14 +74,11 @@ public final class CivModCorePlugin extends ACivMod {
 		registerListener(new ClickableInventoryListener());
 		registerListener(DialogManager.INSTANCE);
 		registerListener(new ScoreBoardListener());
-		registerListener(new CustomEventMapper());
 		registerListener(new WorldTracker());
-		registerListener(ChunkOperationManager.INSTANCE);
 		// Register commands
 		this.commands = new CommandManager(this);
 		this.commands.init();
 		this.commands.registerCommand(new ConfigCommand());
-		this.commands.registerCommand(ChunkOperationManager.INSTANCE);
 		this.commands.registerCommand(new StatCommand());
 		// Load APIs
 		EnchantUtils.loadEnchantAbbreviations(this);
@@ -92,7 +86,6 @@ public final class CivModCorePlugin extends ACivMod {
 		SpawnEggUtils.init();
 		TreeTypeUtils.init();
 		BottomLineAPI.init();
-		MapColours.init();
 		this.skinCache = new SkinCache(this, this.config.getSkinCacheThreads());
 
 		if (this.config.getChunkLoadingStatistics())
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/players/settings/PlayerSettingAPI.java b/src/main/java/vg/civcraft/mc/civmodcore/players/settings/PlayerSettingAPI.java
index 082191b08165e894c97c23efff15d7192cb185f5..dbf5d5e1be70d67748f0562395767d9c163b094a 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/players/settings/PlayerSettingAPI.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/players/settings/PlayerSettingAPI.java
@@ -15,7 +15,6 @@ import org.bukkit.configuration.file.YamlConfiguration;
 import vg.civcraft.mc.civmodcore.CivModCorePlugin;
 import vg.civcraft.mc.civmodcore.players.settings.gui.MenuOption;
 import vg.civcraft.mc.civmodcore.players.settings.gui.MenuSection;
-import vg.civcraft.mc.civmodcore.players.settings.impl.AltConsistentSetting;
 
 /**
  * Allows creating settings, which will automatically be available in players
@@ -85,13 +84,6 @@ public final class PlayerSettingAPI {
 	 */
 	public static void registerSetting(PlayerSetting<?> setting, MenuSection menu) {
 		Preconditions.checkArgument(setting != null, "Player setting cannot be null.");
-		if (setting instanceof AltConsistentSetting) {
-			if (setting.canBeChangedByPlayer()) {
-				menu.addItem(new MenuOption(menu, setting));
-			}
-			menu = null;
-			setting = ((AltConsistentSetting<?,?>) setting).getWrappedSetting();
-		}
 		loadValues(setting);
 		List<PlayerSetting<?>> pluginSettings = SETTINGS_BY_PLUGIN.computeIfAbsent(
 				setting.getOwningPlugin().getName(),
