From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 09:44:10 +0100
Subject: [PATCH] Add parseComponent to ChatUtils


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java
index 8730e4e3bf6a0c829296cbf16cdd248519a4bf93..a2fe7efbfeb264a735852a24afda17e0c77c0095 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java
@@ -210,6 +210,25 @@ public final class ChatUtils {
 		return StringUtils.isBlank(PlainTextComponentSerializer.plainText().serialize(component));
 	}
 
+	// CivPatched Start
+	/**
+	 * Attempts to parse a component from a raw string using MiniMessage unless
+	 * the string contains legacy section formatting.
+	 */
+	public static @Contract("!null -> !null") @Nullable Component parseComponent(final String raw) {
+		if (raw == null) {
+			return null;
+		}
+		if (raw.isBlank()) {
+			return Component.empty();
+		}
+		if (raw.contains("§")) {
+			return LegacyComponentSerializer.legacySection().deserialize(raw);
+		}
+		return MiniMessage.miniMessage().deserialize(raw);
+	}
+	// CivPatched End
+
 	/**
 	 * <p>Determines whether a given base component is null or empty.</p>
 	 *
