From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 14 Jun 2023 03:58:28 +0100
Subject: [PATCH] Add Deferrable and Deregisterable


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java b/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java
index 0cf5ed14fff426640a8b0f0e94c078bc5c5f645f..e761cb1c18a033647b28369a213d889c02c9a5df 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java
@@ -21,7 +21,13 @@ import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.java.JavaPlugin;
 import org.jetbrains.annotations.Contract;
 
-public abstract class ACivMod extends JavaPlugin {
+// CivPatched Start
+import org.jetbrains.annotations.NotNull;
+import vg.civcraft.mc.civmodcore.utilities.defer.Deferrable;
+import vg.civcraft.mc.civmodcore.utilities.defer.DeferredFunctions;
+// CivPatched End
+
+public abstract class ACivMod extends JavaPlugin implements Deferrable { // CivPatched: Implement Deferrable
 
 	private final Set<Class<? extends ConfigurationSerializable>> configClasses = new HashSet<>(0);
 
@@ -43,7 +49,20 @@ public abstract class ACivMod extends JavaPlugin {
 	@Override
 	public void onDisable() {
 		this.configClasses.forEach(ConfigurationSerialization::unregisterClass);
+		getDeferredFunctions().UNSAFE_run(); // CivPatched
+	}
+
+	// CivPatched Start
+	private final DeferredFunctions deferredFunctions = new DeferredFunctions(this);
+	/**
+	 * <p><b>Deferred functions are called during plugin disable.</b></p>
+	 * {@inheritDoc}
+	 */
+	@Override
+	public @NotNull DeferredFunctions getDeferredFunctions() {
+		return this.deferredFunctions;
 	}
+	// CivPatched End
 
 	/**
 	 * Registers a listener class with this plugin.
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/Deferrable.java b/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/Deferrable.java
new file mode 100644
index 0000000000000000000000000000000000000000..88ee7af99eaa00c4d1a6392fcf89ba8b9f46ddf3
--- /dev/null
+++ b/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/Deferrable.java
@@ -0,0 +1,32 @@
+package vg.civcraft.mc.civmodcore.utilities.defer;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This is inspired by Go's anonymous defers. That being said, that's the
+ * extent of the similarity. Do NOT presume that Go-defer rules apply unless
+ * the implementation explicitly says so. Since it's unlikely that implementers
+ * will duplicate {@link #defer(Runnable)} just to update its documentation,
+ * implements are instead encouraged to specify on {@link #getDeferredFunctions()}
+ * exactly when deferred functions are invoked.
+ *
+ * @author CivPatched
+ */
+public interface Deferrable {
+
+    /**
+     * <p>Defers a method to be run once at a later time.</p>
+     *
+     * <p><b>See the documentation for getDeferredFunctions() for information
+     * on when your deferred functions will run.</b></p>
+     */
+    default void defer(final @NotNull Runnable function) {
+        getDeferredFunctions().defer(function);
+    }
+
+    /**
+     * Returns the storage of deferred functions for this deferrable.
+     */
+    @NotNull DeferredFunctions getDeferredFunctions();
+
+}
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/DeferredFunctions.java b/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/DeferredFunctions.java
new file mode 100644
index 0000000000000000000000000000000000000000..a3da356510c6bdd31ee76ae9c3a11740f85bf6f0
--- /dev/null
+++ b/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/DeferredFunctions.java
@@ -0,0 +1,99 @@
+package vg.civcraft.mc.civmodcore.utilities.defer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import lombok.experimental.StandardException;
+import org.bukkit.plugin.java.JavaPlugin;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * @author CivPatched
+ */
+public final class DeferredFunctions implements Deferrable {
+
+    private final Object $lock;
+    private final JavaPlugin owner;
+    private final Logger logger;
+    private final List<Deferred> deferred;
+
+    public DeferredFunctions(
+            final @NotNull JavaPlugin owner
+    ) {
+        this(owner, owner.getLogger());
+    }
+
+    public DeferredFunctions(
+            final @NotNull JavaPlugin owner,
+            final @NotNull Logger logger
+    ) {
+        this.$lock = new Object[0];
+        this.owner = Objects.requireNonNull(owner);
+        this.logger = Objects.requireNonNull(logger);
+        this.deferred = new ArrayList<>(0);
+    }
+
+    public @NotNull JavaPlugin getOwner() {
+        return this.owner;
+    }
+
+    /**
+     * Defers a method to be run once at a later time.
+     */
+    @Override
+    public void defer(
+            final @NotNull Runnable function
+    ) {
+        synchronized (this.$lock) {
+            this.deferred.add(new Deferred(
+                    function,
+                    new DeferredRegistrationStacktrace()
+            ));
+        }
+    }
+
+    @Deprecated
+    @Override
+    public @NotNull DeferredFunctions getDeferredFunctions() {
+        return this;
+    }
+
+    /**
+     * Runs all the deferred functions.
+     */
+    @ApiStatus.Internal
+    public void UNSAFE_run() {
+        this.logger.info("Running deferred functions");
+        final List<Deferred> deferred;
+        synchronized (this.$lock) {
+            deferred = List.copyOf(this.deferred);
+            this.deferred.clear();
+        }
+        for (final Deferred current : deferred) {
+            try {
+                current.function.run();
+            }
+            catch (final Throwable thrown) {
+                this.logger.log(Level.WARNING, "============================================================");
+                this.logger.log(Level.WARNING, "Something went wrong while running a deferred function.");
+                this.logger.log(Level.WARNING, "Here's the registration stacktrace:", current.stacktrace);
+                this.logger.log(Level.WARNING, "Here's the error:", thrown);
+                this.logger.log(Level.WARNING, "============================================================");
+            }
+        }
+    }
+
+    private record Deferred(
+            @NotNull Runnable function,
+            @NotNull Throwable stacktrace
+    ) { }
+
+    @StandardException
+    private static class DeferredRegistrationStacktrace extends Throwable {
+
+    }
+
+}
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/Deregisterable.java b/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/Deregisterable.java
new file mode 100644
index 0000000000000000000000000000000000000000..c1cb13395d9bec8444b79c65cd316b9ce74a453c
--- /dev/null
+++ b/src/main/java/vg/civcraft/mc/civmodcore/utilities/defer/Deregisterable.java
@@ -0,0 +1,13 @@
+package vg.civcraft.mc.civmodcore.utilities.defer;
+
+/**
+ * This is a functional alternative to dedicated deregister methods - it works
+ * very nicely in conjunction with {@link Deferrable}. And since this is an
+ * interface, you can return a tuple-record implementing this if you need to
+ * return additional stuff.
+ */
+public interface Deregisterable {
+
+    void deregister();
+
+}
