From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 10:06:42 +0100
Subject: [PATCH] Remove redundant ACivMod disable cleanups


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java b/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java
index d793b180b06c9073ce595b229b53ad69c6b2adef..6619e06062db8560dde3cb228e4d73991025bc5f 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java
@@ -15,7 +15,6 @@ import org.bukkit.Bukkit;
 import org.bukkit.configuration.serialization.ConfigurationSerializable;
 import org.bukkit.configuration.serialization.ConfigurationSerialization;
 import org.bukkit.event.EventHandler;
-import org.bukkit.event.HandlerList;
 import org.bukkit.event.Listener;
 import org.bukkit.event.server.PluginDisableEvent;
 import org.bukkit.plugin.Plugin;
@@ -43,10 +42,6 @@ public abstract class ACivMod extends JavaPlugin {
 	
 	@Override
 	public void onDisable() {
-		HandlerList.unregisterAll(this);
-		Bukkit.getMessenger().unregisterIncomingPluginChannel(this);
-		Bukkit.getMessenger().unregisterOutgoingPluginChannel(this);
-		Bukkit.getScheduler().cancelTasks(this);
 		this.configClasses.forEach(ConfigurationSerialization::unregisterClass);
 	}
 
