From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 10:22:46 +0100
Subject: [PATCH] Add custom world and console sender contexts


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java b/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java
index 892acaecc9fabfde852b4e697edc001fdd7e74a2..4c33c6a5a34fac5a676f08c428397dc86b350534 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java
@@ -28,6 +28,14 @@ import org.bukkit.plugin.Plugin;
 import vg.civcraft.mc.civmodcore.inventory.items.ItemUtils;
 import vg.civcraft.mc.civmodcore.utilities.CivLogger;
 
+// CivPatched Start
+import co.aikar.commands.InvalidCommandArgument;
+import co.aikar.commands.MinecraftMessageKeys;
+import org.bukkit.World;
+import org.bukkit.command.ConsoleCommandSender;
+import vg.civcraft.mc.civmodcore.utilities.UuidUtils;
+// CivPatched End
+
 /**
  * Command registration class wrapper around {@link BukkitCommandManager}.
  */
@@ -114,6 +122,33 @@ public class CommandManager extends BukkitCommandManager {
 	 *                 {@link #getCommandContexts()}.
 	 */
 	public void registerContexts(@Nonnull final CommandContexts<BukkitCommandExecutionContext> contexts) {
+		// CivPatched Start
+		contexts.registerIssuerAwareContext(ConsoleCommandSender.class, (context) -> {
+			if (context.getSender() instanceof final ConsoleCommandSender console) {
+				return console;
+			}
+			throw new InvalidCommandArgument("Command can only be called from console!", false);
+		});
+		// Override ACF Bukkit's default behaviour of falling back to the sender's world.
+		contexts.registerContext(World.class, (context) -> {
+			final String firstArg = context.getFirstArg();
+			final World world;
+			// Test UUID
+			final UUID worldUUID = UuidUtils.fromString(firstArg);
+			if (worldUUID != null) {
+				world = Bukkit.getWorld(worldUUID);
+			}
+			else {
+				// Otherwise, get from name
+				world = Bukkit.getWorld(firstArg);
+			}
+			if (world != null) {
+				context.popFirstArg();
+				return world;
+			}
+			throw new InvalidCommandArgument(MinecraftMessageKeys.INVALID_WORLD);
+		});
+		// CivPatched End
 	}
 
 	/**
