From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:05:03 +0100
Subject: [PATCH] Update createItemHoverEvent and deprecate asTranslatable


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java
index a2fe7efbfeb264a735852a24afda17e0c77c0095..9a4298c4f5bfd6a724ffd939d9979e43f8bd6602 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/chat/ChatUtils.java
@@ -414,12 +414,7 @@ public final class ChatUtils {
 					.content("<null item>")
 					.build());
 		}
-		return HoverEvent.showItem(
-				item.getType().getKey(),
-				item.getAmount()
-				// TODO: There's a variant of this method that includes an NBT compound. My guess is to include display
-				//       name and lore, perhaps also enchantments, etc..
-		);
+		return item.asHoverEvent(); // CivPatched
 	}
 
 }
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/entities/EntityUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/entities/EntityUtils.java
index 129b5d9686b1fc8fda6985b0f2d6efa06161725e..91d9d275c705c36ef56c3634ff15c0ffa218d24d 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/entities/EntityUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/entities/EntityUtils.java
@@ -42,8 +42,9 @@ public final class EntityUtils {
 	 * @param entityType The entity type to translate.
 	 * @return Returns a translatable component based on the given entity type.
 	 */
+	@Deprecated // CivPatched
 	public static TranslatableComponent asTranslatable(@Nonnull final EntityType entityType) {
-		return Component.translatable(entityType.translationKey());
+		return Component.translatable(entityType); // CivPatched
 	}
 
 }
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/EnchantUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/EnchantUtils.java
index 3264efa21b3703098748e7a967f8d1d5a9049e2c..59749d7b9c7271d5442e43ef2f06ed9ca2d8e6c1 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/EnchantUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/EnchantUtils.java
@@ -108,9 +108,10 @@ public final class EnchantUtils {
 	 * @param enchant The enchantment to get a translatable component for.
 	 * @return Returns a translatable component for the given enchantment.
 	 */
+	@Deprecated // CivPatched
 	@Nonnull
 	public static TranslatableComponent asTranslatable(@Nonnull final Enchantment enchant) {
-		return Component.translatable(enchant.translationKey());
+		return Component.translatable(enchant); // CivPatched
 	}
 
 	/**
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/ItemUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/ItemUtils.java
index 091414d37aec82924aef713676d5bd6f9bee122f..f84ec069bd4bd46c38f473c1eb06d47bce8576c5 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/ItemUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/ItemUtils.java
@@ -28,9 +28,10 @@ public final class ItemUtils {
 	 * @param item The item to get a translatable component for.
 	 * @return Returns a translatable component of the given item.
 	 */
+	@Deprecated // CivPatched
 	@Nonnull
 	public static TranslatableComponent asTranslatable(@Nonnull final ItemStack item) {
-		return Component.translatable(item.translationKey());
+		return Component.translatable(item); // CivPatched
 	}
 
 	/**
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java
index 60959ec9dbc94b0a3e993bf77ecdb79a75ba1c4e..7613b5129e55b275e1122e4e709d8b11aa19d568 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java
@@ -52,9 +52,10 @@ public final class MaterialUtils {
 	 * @param material The material to translate.
 	 * @return Returns a translatable component based on the given material.
 	 */
+	@Deprecated // CivPatched
 	@Nonnull
 	public static TranslatableComponent asTranslatable(@Nonnull final Material material) {
-		return Component.translatable(material.translationKey());
+		return Component.translatable(material); // CivPatched
 	}
 
 	/**
diff --git a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/PotionUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/PotionUtils.java
index 571c3ee4cb70375000ade0afe1dbb31363977e18..730993c16df6187ca63dee7801d1d95d2b622505 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/PotionUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/PotionUtils.java
@@ -7,7 +7,6 @@ import javax.annotation.Nullable;
 import lombok.experimental.UtilityClass;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.TranslatableComponent;
-import net.minecraft.world.effect.MobEffect;
 import org.apache.commons.lang3.tuple.Pair;
 import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
@@ -56,23 +55,10 @@ public final class PotionUtils {
 	 * @param effect The potion effect to get a translatable component for.
 	 * @return Returns a translatable component for the given potion effect.
 	 */
-	@SuppressWarnings("deprecation")
+	@Deprecated // CivPatched
 	@Nonnull
 	public static TranslatableComponent asTranslatable(@Nonnull final PotionEffectType effect) {
-		final MobEffect mojang = MobEffect.byId(effect.getId());
-		assert mojang != null;
-		final String key = mojang.getDescriptionId(); // Gets the translation key
-		// If the obfuscation has changed, look for a method on MobEffectList like:
-		//
-		//    protected String b() {
-		//        if (this.d == null) {
-		//            this.d = SystemUtils.a("effect", IRegistry.V.getKey(this));
-		//        }
-		//        return this.d;
-		//    }
-		//
-		// then use whatever method calls that method
-		return Component.translatable(key);
+		return Component.translatable(effect); // CivPatched
 	}
 
 	/**
