From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:11:05 +0100
Subject: [PATCH] Add DeprecationUtils


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/utilities/DeprecationUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/utilities/DeprecationUtils.java
new file mode 100644
index 0000000000000000000000000000000000000000..5c0f9129de318e5f79838c9cc9ad1d15ceeb5d5a
--- /dev/null
+++ b/src/main/java/vg/civcraft/mc/civmodcore/utilities/DeprecationUtils.java
@@ -0,0 +1,31 @@
+package vg.civcraft.mc.civmodcore.utilities;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import lombok.experimental.StandardException;
+import lombok.experimental.UtilityClass;
+import org.bukkit.Bukkit;
+
+/**
+ * @author CivPatched
+ */
+@UtilityClass
+public class DeprecationUtils {
+
+    private final Logger LOGGER = Bukkit.getLogger();
+
+    /**
+     * Prints a warning to Bukkit's console with a stacktrace warning of deprecation usages.
+     */
+    public void printDeprecationWarning() {
+        LOGGER.log(Level.WARNING,
+                "You are using a deprecated feature. Please consider updating to its recommended replacement.",
+                new DeprecationException());
+    }
+
+    @StandardException
+    private static class DeprecationException extends RuntimeException {
+
+    }
+
+}
\ No newline at end of file
