From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:17:16 +0100
Subject: [PATCH] Add cancelPrepareItemCraftEvent to EventUtils


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/events/EventUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/events/EventUtils.java
index bb45cc7787f5a6aa3d282f665a4238758b721107..b500ffc3f08307ae709d08b29d9ce549587cfc2e 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/events/EventUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/events/EventUtils.java
@@ -5,6 +5,10 @@ import org.bukkit.event.player.PlayerInteractEvent;
 import org.bukkit.inventory.ItemStack;
 import org.jetbrains.annotations.NotNull;
 
+// CivPatched Start
+import org.bukkit.event.inventory.PrepareItemCraftEvent;
+// CivPatched End
+
 public final class EventUtils {
 
 	/**
@@ -19,4 +23,16 @@ public final class EventUtils {
 		event.getPlayer().getInventory().setItem(Objects.requireNonNull(event.getHand()), item);
 	}
 
+	// CivPatched Start
+	/**
+	 * Given that {@link PrepareItemCraftEvent} cannot be cancelled in the traditional sense, we need to instead
+	 * nullify its result.
+	 *
+	 * @param event The prepare event.
+	 */
+	public void cancelPrepareItemCraftEvent(final @NotNull PrepareItemCraftEvent event) {
+		event.getInventory().setResult(null);
+	}
+	// CivPatched End
+
 }
