From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:22:41 +0100
Subject: [PATCH] Standardise NBTSerializationException


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/nbt/NBTSerializationException.java b/src/main/java/vg/civcraft/mc/civmodcore/nbt/NBTSerializationException.java
index 8b4eb90deac0e9fc34c62448177b3e7fad6d49bc..3586d1e72aee2fa9b5b16e752131a3ad66e344e8 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/nbt/NBTSerializationException.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/nbt/NBTSerializationException.java
@@ -7,25 +7,10 @@ import vg.civcraft.mc.civmodcore.nbt.wrappers.NBTCompound;
  * Exception that ought to be used within {@link NBTSerializable#toNBT(NBTCompound)} and
  * {@link NBTSerializable#fromNBT(NBTCompound)}.
  */
+@lombok.experimental.StandardException // CivPatched
 public class NBTSerializationException extends RuntimeException {
 
 	@Serial
 	private static final long serialVersionUID = 606023177729327630L;
 
-	public NBTSerializationException() {
-		super();
-	}
-
-	public NBTSerializationException(String message) {
-		super(message);
-	}
-
-	public NBTSerializationException(String message, Throwable cause) {
-		super(message, cause);
-	}
-
-	public NBTSerializationException(Throwable cause) {
-		super(cause);
-	}
-
 }
\ No newline at end of file
