From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:26:35 +0100
Subject: [PATCH] Add MoreIteratorUtils


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/utilities/MoreIteratorUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/utilities/MoreIteratorUtils.java
new file mode 100644
index 0000000000000000000000000000000000000000..3f4f7532303aa1c2e45041e0dde6b537d3a89387
--- /dev/null
+++ b/src/main/java/vg/civcraft/mc/civmodcore/utilities/MoreIteratorUtils.java
@@ -0,0 +1,33 @@
+package vg.civcraft.mc.civmodcore.utilities;
+
+import java.util.Iterator;
+import java.util.function.Predicate;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.collections4.IteratorUtils;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Utility class that fills in the gaps of {@link IteratorUtils}.
+ *
+ * @author CivPatched
+ */
+@UtilityClass
+public class MoreIteratorUtils {
+
+    /**
+     * @param <T> The iterator's element type.
+     * @param iterator The iterator to remove elements from.
+     * @param predicate The method to test entries with. It should return true to remove that entry.
+     */
+    public <T> void removeIf(
+            final @NotNull Iterator<T> iterator,
+            final @NotNull Predicate<T> predicate
+    ) {
+        while (iterator.hasNext()) {
+            if (predicate.test(iterator.next())) {
+                iterator.remove();
+            }
+        }
+    }
+
+}
\ No newline at end of file
