From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:24:34 +0100
Subject: [PATCH] Add getMaterials to MaterialUtils


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java
index 7613b5129e55b275e1122e4e709d8b11aa19d568..8e22ae05e1c631c91f57fb5ddf216dacee72fef8 100644
--- a/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java
+++ b/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/MaterialUtils.java
@@ -14,6 +14,12 @@ import org.bukkit.Material;
 import org.bukkit.Tag;
 import org.bukkit.inventory.ItemStack;
 
+// CivPatched Start
+import java.util.HashSet;
+import java.util.Set;
+import org.jetbrains.annotations.NotNull;
+// CivPatched End
+
 /**
  * <p>See <a href="https://github.com/Protonull/BukkitReport/tree/master/reports">BukkitReports</a>.</p>
  *
@@ -34,6 +40,18 @@ public final class MaterialUtils {
 		addAll(MaterialTags.CONCRETES.getValues());
 	}};
 
+	// CivPatched Start
+	public final Set<Material> MATERIALS = Set.of(Material.values());
+	/**
+	 * @return Returns a new material set. DO NOT USE {@link java.util.EnumSet<Material>}! As it'll force load all
+	 * materials, including all legacy materials, forcing the server into legacy mode. For some reason
+	 * {@link Material#values()} doesn't do that... odd.
+	 */
+	public @NotNull Set<Material> getMaterials() {
+		return new HashSet<>(MATERIALS);
+	}
+	// CivPatched End
+
 	/**
 	 * Attempts to retrieve a material by its slug.
 	 *
