From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 13 May 2023 18:27:15 +0100
Subject: [PATCH] Add MoreIterableUtils


diff --git a/src/main/java/vg/civcraft/mc/civmodcore/utilities/MoreIterableUtils.java b/src/main/java/vg/civcraft/mc/civmodcore/utilities/MoreIterableUtils.java
new file mode 100644
index 0000000000000000000000000000000000000000..5b78ccd5dc9e9bea27a06571e6938e74bb723762
--- /dev/null
+++ b/src/main/java/vg/civcraft/mc/civmodcore/utilities/MoreIterableUtils.java
@@ -0,0 +1,28 @@
+package vg.civcraft.mc.civmodcore.utilities;
+
+import java.util.function.Predicate;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.collections4.IterableUtils;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Utility class that fills in the gaps of {@link IterableUtils}.
+ *
+ * @author CivPatched
+ */
+@UtilityClass
+public class MoreIterableUtils {
+
+    /**
+     * @param <T> The iterable's element type.
+     * @param iterable The iterable to remove elements from.
+     * @param predicate The method to test entries with. It should return true to remove that entry.
+     */
+    public <T> void removeIf(
+            final @NotNull Iterable<T> iterable,
+            final @NotNull Predicate<T> predicate
+    ) {
+        MoreIteratorUtils.removeIf(iterable.iterator(), predicate);
+    }
+
+}
\ No newline at end of file
