From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Thu, 18 May 2023 01:01:06 +0100
Subject: [PATCH] Update canvas to 8b25285839


diff --git a/build.gradle.kts b/build.gradle.kts
index e26fefa23071ee21936262f2d4d2724c727acafa..5d4fdb0ca86c756b99a1ecc3dc96709337caf0da 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -10,7 +10,7 @@ dependencies {
 	api("com.zaxxer:HikariCP:5.0.1")
 	api("co.aikar:acf-bukkit:0.5.0-SNAPSHOT")
 	api("co.aikar:taskchain-bukkit:3.7.2")
-	api("com.github.IPVP-MC:canvas:91ec97f076")
+	api("com.github.IPVP-MC:canvas:8b25285839")
 	api("org.apache.commons:commons-collections4:4.4")
 	api("com.google.code.findbugs:jsr305:3.0.2")
 }
diff --git a/src/main/java/org/ipvp/canvas/type/AbstractCivMenu.java b/src/main/java/org/ipvp/canvas/type/AbstractCivMenu.java
index df544310700fe00df8889b3179edf9dd3fb86c93..064ffa45f5607f4716fc9ea26843baeb026c68dd 100644
--- a/src/main/java/org/ipvp/canvas/type/AbstractCivMenu.java
+++ b/src/main/java/org/ipvp/canvas/type/AbstractCivMenu.java
@@ -29,7 +29,7 @@ public abstract class AbstractCivMenu extends AbstractMenu {
 	private static final Method UPDATE_INV_CONTENTS_METHOD;
 
 	static {
-		VIEWERS_FIELD = FieldUtils.getDeclaredField(AbstractMenu.class, "viewers", true);
+		VIEWERS_FIELD = FieldUtils.getDeclaredField(AbstractMenu.class, "holders", true); // CivPatched: The field was renamed to "holders"
 		SLOTS_FIELD = FieldUtils.getDeclaredField(AbstractMenu.class, "slots", true);
 		UPDATE_INV_CONTENTS_METHOD = MethodUtils.getMatchingMethod(AbstractMenu.class,
 				"updateInventoryContents", Player.class, Inventory.class);
