From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 14 Jun 2023 08:26:41 +0100
Subject: [PATCH] Fixup ItemExchange


diff --git a/build.gradle.kts b/build.gradle.kts
index 506d5dbca920b621b8d75e9b08ee2ae28bce6124..efa6f3b2565c2d458de226b986b3a32bbad3f569 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,54 +1,14 @@
-import net.civmc.civgradle.common.util.civRepo
+group = "civ.genre.itemexchange"
+version = "2.0.1"
+description = "ItemExchange"
 
-plugins {
-    `java-library`
-    `maven-publish`
-    id("net.civmc.civgradle.plugin") version "1.0.0-SNAPSHOT"
+configure<net.civmc.civgradle.CivGradleExtension> {
+	pluginName = project.property("description") as String
 }
 
-// Temporary hack:
-// Remove the root build directory
-gradle.buildFinished {
-	project.buildDir.deleteRecursively()
-}
-
-allprojects {
-	group = "net.civmc.itemexchange"
-	version = "2.0.1"
-	description = "ItemExchange"
-}
-
-subprojects {
-	apply(plugin = "net.civmc.civgradle.plugin")
-	apply(plugin = "java-library")
-	apply(plugin = "maven-publish")
-
-	java {
-		toolchain {
-			languageVersion.set(JavaLanguageVersion.of(17))
-		}
-	}
-
-	repositories {
-		mavenCentral()
-        civRepo("CivMC/CivModCore")
-	}
-
-	publishing {
-		repositories {
-			maven {
-				name = "GitHubPackages"
-				url = uri("https://maven.pkg.github.com/CivMC/ItemExchange")
-				credentials {
-					username = System.getenv("GITHUB_ACTOR")
-					password = System.getenv("GITHUB_TOKEN")
-				}
-			}
-		}
-		publications {
-			register<MavenPublication>("gpr") {
-				from(components["java"])
-			}
-		}
-	}
+dependencies {
+	compileOnly(project(":CivModCore"))
+	compileOnly(project(":NameLayer"))
+	compileOnly("net.civmc.citadel:citadel-paper:5.0.0:dev")
+	compileOnly(project(":JukeAlert"))
 }
diff --git a/src/main/java/com/untamedears/itemexchange/utility/Utilities.java b/src/main/java/com/untamedears/itemexchange/utility/Utilities.java
index a4d0bce0dd38820ede18615e4e6b3e0c3fd8165c..89a99229f1e5607cbb7dcb0fda6cbd7a6410b4ff 100644
--- a/src/main/java/com/untamedears/itemexchange/utility/Utilities.java
+++ b/src/main/java/com/untamedears/itemexchange/utility/Utilities.java
@@ -11,7 +11,6 @@ import java.util.Map;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import javax.annotation.Nonnull;
-import lombok.experimental.ExtensionMethod;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.collections4.MapUtils;
 import org.apache.commons.lang3.ArrayUtils;
@@ -29,7 +28,6 @@ import org.bukkit.inventory.ItemStack;
 import org.bukkit.potion.PotionData;
 import org.bukkit.potion.PotionEffect;
 import vg.civcraft.mc.civmodcore.inventory.InventoryUtils;
-import vg.civcraft.mc.civmodcore.utilities.JavaExtensions;
 import vg.civcraft.mc.civmodcore.utilities.KeyedUtils;
 import vg.civcraft.mc.civmodcore.utilities.NullUtils;
 import vg.civcraft.mc.civmodcore.world.WorldUtils;
@@ -37,7 +35,6 @@ import vg.civcraft.mc.civmodcore.world.WorldUtils;
 /**
  * A series of Utilities of ItemExchange
  */
-@ExtensionMethod(JavaExtensions.class)
 public final class Utilities {
 
 	/**
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
index 73d0434f15dc8c21f83344454ee478ac8a8ed0aa..b39055373f8eaf89d7af2550c7fee3cdac38f5e7 100644
--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -2,7 +2,7 @@ name: ${name}
 version: ${version}
 authors: [gmlaxfanatic, Protonull]
 main: com.untamedears.itemexchange.ItemExchangePlugin
-api-version: 1.16
+api-version: 1.18
 depend: [CivModCore]
 softdepend: [NameLayer, Citadel, JukeAlert]
 permissions:
