From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 14 Jun 2023 08:26:41 +0100
Subject: [PATCH] Fixup ItemExchange


diff --git a/build.gradle.kts b/build.gradle.kts
new file mode 100644
index 0000000000000000000000000000000000000000..dce4ee25eaeefa18b72f13e387d31a143a3ee53c
--- /dev/null
+++ b/build.gradle.kts
@@ -0,0 +1,14 @@
+group = "civ.genre.itemexchange"
+version = "2.0.1"
+description = "ItemExchange"
+
+configure<net.civmc.civgradle.CivGradleExtension> {
+	pluginName = project.property("description") as String
+}
+
+dependencies {
+	compileOnly(project(":CivModCore"))
+	compileOnly(project(":NameLayer"))
+	compileOnly(project(":Citadel"))
+	compileOnly(project(":JukeAlert"))
+}
diff --git a/src/main/java/com/untamedears/itemexchange/utility/ModifierStorage.java b/src/main/java/com/untamedears/itemexchange/utility/ModifierStorage.java
index a27fafe9a54a999182ba2cf16c7dddb39a52f189..fe3f9cd5c55d7f7f013e6bdf39e28e4491ac49dc 100644
--- a/src/main/java/com/untamedears/itemexchange/utility/ModifierStorage.java
+++ b/src/main/java/com/untamedears/itemexchange/utility/ModifierStorage.java
@@ -3,7 +3,6 @@ package com.untamedears.itemexchange.utility;
 import com.google.common.collect.ClassToInstanceMap;
 import com.google.common.collect.MutableClassToInstanceMap;
 import com.untamedears.itemexchange.rules.interfaces.ModifierData;
-import com.untamedears.itemexchange.utility.functional.NonNullSupplier;
 import java.util.Iterator;
 import java.util.Objects;
 import java.util.Spliterator;
@@ -13,6 +12,11 @@ import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import lombok.EqualsAndHashCode;
 
+// CivPatched Start
+import java.util.function.Supplier;
+import org.jetbrains.annotations.NotNull;
+// CivPatched End
+
 /**
  * A storage method for modifiers that puts significance on the modifier's class.
  */
@@ -99,7 +103,7 @@ public final class ModifierStorage implements Iterable<ModifierData> {
 	@Nonnull
 	@SuppressWarnings("unchecked")
 	public <T extends ModifierData> T getOrDefault(@Nonnull final Class<T> clazz,
-												   @Nonnull final NonNullSupplier<T> supplier) {
+												   @NotNull final Supplier<@NotNull T> supplier) { // CivPatched
 		return (T) this.map.computeIfAbsent(Objects.requireNonNull(clazz), (_clazz) -> supplier.get());
 	}
 
@@ -115,7 +119,7 @@ public final class ModifierStorage implements Iterable<ModifierData> {
 	@Nonnull
 	@SuppressWarnings("unchecked")
 	public <T extends ModifierData> T getOrDefault(@Nonnull final T instance,
-												   @Nonnull final NonNullSupplier<T> supplier) {
+												   @NotNull final Supplier<@NotNull T> supplier) { // CivPatched
 		return getOrDefault((Class<T>) instance.getClass(), supplier);
 	}
 
diff --git a/src/main/java/com/untamedears/itemexchange/utility/Utilities.java b/src/main/java/com/untamedears/itemexchange/utility/Utilities.java
index a4d0bce0dd38820ede18615e4e6b3e0c3fd8165c..89a99229f1e5607cbb7dcb0fda6cbd7a6410b4ff 100644
--- a/src/main/java/com/untamedears/itemexchange/utility/Utilities.java
+++ b/src/main/java/com/untamedears/itemexchange/utility/Utilities.java
@@ -11,7 +11,6 @@ import java.util.Map;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import javax.annotation.Nonnull;
-import lombok.experimental.ExtensionMethod;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.collections4.MapUtils;
 import org.apache.commons.lang3.ArrayUtils;
@@ -29,7 +28,6 @@ import org.bukkit.inventory.ItemStack;
 import org.bukkit.potion.PotionData;
 import org.bukkit.potion.PotionEffect;
 import vg.civcraft.mc.civmodcore.inventory.InventoryUtils;
-import vg.civcraft.mc.civmodcore.utilities.JavaExtensions;
 import vg.civcraft.mc.civmodcore.utilities.KeyedUtils;
 import vg.civcraft.mc.civmodcore.utilities.NullUtils;
 import vg.civcraft.mc.civmodcore.world.WorldUtils;
@@ -37,7 +35,6 @@ import vg.civcraft.mc.civmodcore.world.WorldUtils;
 /**
  * A series of Utilities of ItemExchange
  */
-@ExtensionMethod(JavaExtensions.class)
 public final class Utilities {
 
 	/**
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
index 73d0434f15dc8c21f83344454ee478ac8a8ed0aa..b39055373f8eaf89d7af2550c7fee3cdac38f5e7 100644
--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -2,7 +2,7 @@ name: ${name}
 version: ${version}
 authors: [gmlaxfanatic, Protonull]
 main: com.untamedears.itemexchange.ItemExchangePlugin
-api-version: 1.16
+api-version: 1.18
 depend: [CivModCore]
 softdepend: [NameLayer, Citadel, JukeAlert]
 permissions:
